hi-ml/src/health_ml/model_trainer.py

```diff
@@ -255,6 +260,17 @@ def model_train(checkpoint_path: Optional[Path],
     assert trainer.logger is not None
     trainer.logger.finalize('success')
 
+    if container.additional_val_epoch and hasattr(lightning_model, "additional_val_epoch"):
+        if checkpoint_handler:
+            checkpoint_handler.additional_training_done()
```

Comment:
I wasn't sure if this should be called here as well to get the right checkpoint. I am planning on refactoring this anyways to avoid loading the checkpoint twice for the extra val epoch and test but in a seperate PR as we discussed last time (enabling running inference in DDP with the same trainer instance).